src/agent0/chainsync/db/hyperdrive/chain_to_db.py

```diff
@@ -1,149 +1,121 @@
+
+from .convert_data import convert_checkpoint_events, convert_pool_config, convert_pool_info, convert_trade_events
 from .interface import (
-    add_checkpoint_info,
     add_pool_config,
     add_pool_infos,
-    add_transactions,
-    add_wallet_deltas,
+    get_latest_block_number_from_checkpoint_info_table,
     get_latest_block_number_from_trade_event,
 )
-from .schema import TradeEvent
+from .schema import CheckpointInfo, TradeEvent
 
 
 def init_data_chain_to_db(
-    interface: HyperdriveReadInterface,
+    interfaces: list[HyperdriveReadInterface],
     session: Session,
 ) -> None:
     """Function to query and insert pool config to dashboard.
 
     Arguments
     ---------
-    interface: HyperdriveReadInterface
-        The hyperdrive interface object.
+    interfaces: list[HyperdriveReadInterface]
+        A collection of Hyperdrive interface objects, each connected to a pool.
     session: Session
         The database session
     """
-    pool_config_dict = asdict(interface.current_pool_state.pool_config)
-    pool_config_dict["hyperdrive_address"] = interface.hyperdrive_address
-    fees = pool_config_dict["fees"]
-    pool_config_dict["curve_fee"] = fees["curve"]
-    pool_config_dict["flat_fee"] = fees["flat"]
-    pool_config_dict["governance_lp_fee"] = fees["governance_lp"]
-    pool_config_dict["governance_zombie_fee"] = fees["governance_zombie"]
-    pool_config_dict["inv_time_stretch"] = FixedPoint(1) / pool_config_dict["time_stretch"]
-    pool_config_db_obj = convert_pool_config(pool_config_dict)
-    add_pool_config(pool_config_db_obj, session)
-
-
-def data_chain_to_db(interface: HyperdriveReadInterface, block: BlockData, session: Session) -> None:
+    for interface in interfaces:
+        pool_config_dict = asdict(interface.current_pool_state.pool_config)
+        pool_config_dict["hyperdrive_address"] = interface.hyperdrive_address
+        fees = pool_config_dict["fees"]
+        pool_config_dict["curve_fee"] = fees["curve"]
+        pool_config_dict["flat_fee"] = fees["flat"]
+        pool_config_dict["governance_lp_fee"] = fees["governance_lp"]
+        pool_config_dict["governance_zombie_fee"] = fees["governance_zombie"]
+        pool_config_dict["inv_time_stretch"] = FixedPoint(1) / pool_config_dict["time_stretch"]
+        pool_config_db_obj = convert_pool_config(pool_config_dict)
+        add_pool_config(pool_config_db_obj, session)
+
+
+def data_chain_to_db(interfaces: list[HyperdriveReadInterface], block_number: int, session: Session) -> None:
     """Function to query and insert data to dashboard.
 
     Arguments
     ---------
-    interface: HyperdriveReadInterface
-        Interface for the market on which this agent will be executing trades (MarketActions).
-    block: BlockData
-        The block to query.
+    interfaces: list[HyperdriveReadInterface]
+        A collection of Hyperdrive interface objects, each connected to a pool.
+    block_number: int
+        The block number to data on.
```

Comment:
> The block number to data on.

is a weird sentence lol. "The block number associated with the query." or "The query is filtered on this block number." or something?

Reply:
I blame codeium for generating this